File: redact-dropwizard-example/src/main/java/us/jsy/redact/pii/SSN.java
package us.jsy.redact.pii;

import us.jsy.redact.RedactJackson;
import us.jsy.redact.Strategy;

import java.util.function.Function;

public class SSN extends RedactJackson {
    public SSN(String pii) {
        super(pii);
    }

    @Override
    protected Function<String, String> strategy() {
        return Strategy.LAST_4;
    }
}

File: redact-dropwizard-kotlin-example/src/main/java/us/jsy/redact/pii/SSN.kt
package us.jsy.redact.pii

import us.jsy.redact.RedactJackson
import us.jsy.redact.Strategy
import java.util.function.Function

class SSN(pii: String?) : RedactJackson(pii) {
    override fun strategy(): Function<String, String> {
        return Strategy.LAST_4
    }
}